website/docs/externaldata.md

```diff
@@ -430,3 +430,68 @@ server := &http.Server{
 ```
 
 2. If `cert-controller` is disabled via the `--disable-cert-rotation` flag, you can use a cluster-wide, well-known CA certificate for Gatekeeper so that your external data provider can trust it without being deployed to the `gatekeeper-system` namespace.
+
+### Authenticate API server against Webhook (Self managed kube cluster only)
+
+To have requests coming from apiserver authenticated by webhook, following configuration can be made:
+
+1. Deploy gatekeeper a client CA name. Provide name of the client ca with the flag `--client-cert-name`. The same name will be used to read certificate from the webhook secret. The webhook will only authenticate apiserver requests if both client ca name is provided with flag.
+
+2. You will need to patch the webhook secret manually to attach client ca crt. Update secret `gatekeeper-webhook-server-cert` to include `clientca.crt`. Key name `clientca.crt` should match the name passed with `--client-cert-name` flag.
+
+    ```
+    kind: Secret
+    apiVersion: v1
+    data:
+      ca.crt: <ca-crt>
+      ca.key: <ca-key>
+      tls.crt: <tls-crt>
+      tls.key: <tls-key>
+      clientca.crt: <apiserver's ca.crt>
+    metadata:
+      ...
+      name: gatekeeper-webhook-server-cert
+      namespace: gatekeeper-system
+    type: Opaque
+    ```
+3. You will need to make sure that apiserver includes appropriate certificate while sending requests to apiserver, otherwise webhook will not accept these requests and will log error of `tls client didn't provide a certificate`. To make sure apiserver attaches correct certificate to requests being sent to webhook, you must specify the location of the admission control configuration file via the `--admission-control-config-file` flag while starting apiserver. Here is an example admission control configuration file:
+    ```
+    apiVersion: apiserver.config.k8s.io/v1
+    kind: AdmissionConfiguration
+    plugins:
+    - name: ValidatingAdmissionWebhook
+      configuration:
+        apiVersion: apiserver.config.k8s.io/v1
+        kind: WebhookAdmissionConfiguration
+        kubeConfigFile: "<path-to-kubeconfig-file>"
+    - name: MutatingAdmissionWebhook
+      configuration:
+        apiVersion: apiserver.config.k8s.io/v1
+        kind: WebhookAdmissionConfiguration
+        kubeConfigFile: "<path-to-kubeconfig-file>"
+    ```
+  
+    KubeConfig file should look something like:
+
+    ```
+    apiVersion: v1
+    clusters:
+    - cluster:
+        certificate-authority-data: <clientca.crt> # same value as provided in gatekeeper webhook secret's clientca.crt 
+        server: https://gatekeeper-webhook-service.gatekeeper-system.svc:443
```

Comment:
can we add note about service or namespace name and it should be updated if it's different